src/Handler/Table/Preview/PreviewTableHandler.php

```diff
@@ -0,0 +1,166 @@
+            // TODO fulltextSearch
+            // TODO whereFilters
+            $selectTableSql = sprintf(
+                "SELECT %s %s\nFROM %s.%s",
+                $command->getLimit() ? sprintf('TOP %d', $command->getLimit()) : '',
+                implode(', ', array_map([TeradataQuote::class, 'quoteSingleIdentifier'], $columns)),
+                TeradataQuote::quoteSingleIdentifier($databaseName),
+                TeradataQuote::quoteSingleIdentifier($command->getTableName())
+            );
+
+            if ($command->hasOrderBy() && $command->getOrderBy()) {
+                $orderBy = $command->getOrderBy();
+                $selectTableSql .= sprintf(
+                    "\nORDER BY %s %s",
+                    // TODO k cemu je potreba dataType?
+                    TeradataQuote::quoteSingleIdentifier($orderBy->getColumnName()),
+                    $orderBy->getOrder() === PreviewTableCommand\PreviewTableOrderBy\Order::ASC ? 'ASC' : 'DESC'
+                );
+            }
+
+            // select table
+            $result = $db->executeQuery($selectTableSql);
+
+            // get columns definitions
+            $tableRef = new TeradataTableReflection($db, $databaseName, $command->getTableName());
+            /** @var TeradataColumn[] $columnsDefArr */
+            $columnsDefArr = iterator_to_array($tableRef->getColumnsDefinitions());
+            /** @var array<string, Teradata> $columnsDef */
+            $columnsDef = [];
+            foreach ($columnsDefArr as $index => $columnDef) {
+                $columnsDef[$columnDef->getColumnName()] = $columnDef->getColumnDefinition();
+            }
+
+            // set response
+            $response = new PreviewTableResponse();
+
+            $rows = new RepeatedField(GPBType::MESSAGE, PreviewTableResponse\Row::class);
+            foreach ($result->fetchAllAssociative() as $lineNumber => $line) {
+                // set column names
+                if ($lineNumber === 0) {
+                    $columns = new RepeatedField(GPBType::STRING);
+                    foreach (array_keys($line) as $column) {
+                        $columns[] = $column;
+                    }
+                    $response->setColumns($columns);
+                }
+
+                // set row
+                $row = new PreviewTableResponse\Row();
+                $columns = new RepeatedField(GPBType::MESSAGE, PreviewTableResponse\Row\Column::class);
+                foreach ($line as $itemKey => $itemValue) {
+                    // set row columns
+                    $value = new Value();
+                    $truncated = false;
+                    if ($itemValue === null) {
+                        $value->setNullValue(NullValue::NULL_VALUE);
+                    } else {
+                        // preview returns all data as string
+                        switch ($columnsDef[$itemKey]->getBasetype()) {
+                            case BaseType::NUMERIC:
```

Comment:
boolean ne proc? a pokud je to spravne, tak tato konstrukce mi prijde takova divoka. Spis `if($columnsDef[$itemKey]->getBasetype() != BOOLEAN)` nebo tak neco ne? V cem ma toto vyhodu? Basetype nebudeme snad nikdy menit

Reply:
jo, puvodne jsem tam pracoval s vice typy - upravim...